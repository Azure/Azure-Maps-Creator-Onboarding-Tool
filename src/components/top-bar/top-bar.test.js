import { render, screen } from '@testing-library/react';

import TopBar from './top-bar';

jest.mock('hooks', () => ({
  useFeatureFlags: () => ({ isPlacesPreview: false }),
}));

describe('TopBar', () => {
  it('should match snapshot', () => {
    expect(render(<TopBar />)).toMatchSnapshot();
  });

  it('should render', () => {
    render(<TopBar />);
    const textSpanElement0 = screen.getByText(/Microsoft Azure/i);
    const textSpanElement1 = screen.getByText(/Azure Maps Creator/i);

    const elements = [textSpanElement0, textSpanElement1];

    elements.forEach(textElement => {
      expect(textElement).toBeInTheDocument();
    });
  });
});
