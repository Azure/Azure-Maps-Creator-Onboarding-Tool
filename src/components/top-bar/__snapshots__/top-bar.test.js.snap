// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  background: #0078d4;
  color: white;
  height: 2.5rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  box-shadow: 0px 0px 0.625rem #888888;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  font-weight: 600;
  font-size: 0.938rem;
  margin: 0 1.25rem;
}

.emotion-3 {
  width: 1px;
  height: 1.25rem;
  background: white;
}

.emotion-4 {
  font-size: 0.875rem;
  margin: 0 1.25rem;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1rem;
  margin-right: 1.25rem;
}

.emotion-6 {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-6:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<body>
    <div>
      <div
        class="emotion-0"
      >
        <div
          class="emotion-1"
        >
          <span
            class="emotion-2"
          >
            Microsoft Azure
          </span>
          <span
            class="emotion-3"
          />
          <span
            class="emotion-4"
          >
            Azure Maps Creator
          </span>
        </div>
        <div
          class="emotion-5"
        >
          <a
            aria-label="docs.link"
            class="emotion-6"
            href="https://learn.microsoft.com/en-us/azure/azure-maps/drawing-package-guide?pivots=drawing-package-v2"
            rel="noreferrer"
            target="_blank"
          >
            docs
          </a>
          <a
            aria-label="feedback.link"
            class="emotion-6"
            href="https://feedback.azure.com/d365community/post/fc834083-0925-ec11-b6e6-000d3a4f09d0?page=1&sort=newest"
            rel="noreferrer"
            target="_blank"
          >
            feedback
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  background: #0078d4;
  color: white;
  height: 2.5rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  box-shadow: 0px 0px 0.625rem #888888;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  font-weight: 600;
  font-size: 0.938rem;
  margin: 0 1.25rem;
}

.emotion-3 {
  width: 1px;
  height: 1.25rem;
  background: white;
}

.emotion-4 {
  font-size: 0.875rem;
  margin: 0 1.25rem;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1rem;
  margin-right: 1.25rem;
}

.emotion-6 {
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-6:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-1"
      >
        <span
          class="emotion-2"
        >
          Microsoft Azure
        </span>
        <span
          class="emotion-3"
        />
        <span
          class="emotion-4"
        >
          Azure Maps Creator
        </span>
      </div>
      <div
        class="emotion-5"
      >
        <a
          aria-label="docs.link"
          class="emotion-6"
          href="https://learn.microsoft.com/en-us/azure/azure-maps/drawing-package-guide?pivots=drawing-package-v2"
          rel="noreferrer"
          target="_blank"
        >
          docs
        </a>
        <a
          aria-label="feedback.link"
          class="emotion-6"
          href="https://feedback.azure.com/d365community/post/fc834083-0925-ec11-b6e6-000d3a4f09d0?page=1&sort=newest"
          rel="noreferrer"
          target="_blank"
        >
          feedback
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
